### PR TITLE
Add JacksonSerializer to serialize IntOrString

### PR DESCRIPTION
**Pull request by wkclz**
It's OK when using Gson, but not all of the project want to dependency Gson when jackson is the default serialize tool. If return null instead of an exception, it can work well with jackson or other serialize tool.

### CODE REVIEW

**Comment by brendandburns** [comment]
Can you explain more where the problem is? Neither of these methods should be involved in deserialization as far as I know.

Perhaps add a unit test that demonstrates the issue?

**Comment by wkclz** [comment]
问题在于，序列化工具会使用getter来序列化任意字段，也许也不想序列化它。当序列化到 IntOrString 时，就抛出异常了。
The problem is, serialization tool would serialize any field with getter even if you don't want to serialize it.. When we serialize an object witch contant IntOrString, we got the exception.

这里写了一个 Gson 的序列化适配，但对于Jackson (Spring框架的默认序列化工具)，却没法适配，所以可以找到 34个由此导致的 issue.
Here is an adapter with Gson, but not for Jackson, Jackson is the default serialization tool in spring framework. So we got the issues like: #2593, #2566, (34 issues). 

我没搞明白为什么这里有一个不明确类型的类，也许它本应该是一个字符串，在需要它时，再转换为数字。
I don't understand why there is such an ambiguous type, perhaps it should always be a string, and then judge whether it should be a number when needed.

**Comment by wkclz** [comment]
demo: #2566

**Comment by brendandburns** [comment]
IntOrString is part of the Kubernetes spec, so it's needed. The API server chooses to interpret it is either a string or an int and so we really can't do anything other than the way it is currently done. The Kubernetes API server sees `"1"` as different than `1`.

I think that you need to write a custom serialization adapter for Jackson:

https://www.baeldung.com/jackson-custom-serialization

**Comment by wkclz** [comment]
It's okay now.

**Comment by wkclz** [comment]
但我仍然认为不应该设置一个不准确的返回类型给序列化工具，而导致增加额外的工作去适配，尤其是抛出异常的方式。
> But I still believe that we should not give an inaccurate type to serialization tools, in particular an Exception, it would make us more workload.

后续优化建议这两个方向，看是否合理：
> Here are two suggestions if reasonable:

- 在尝试获取错误的类型时，返回空而不是抛出异常，会对序列化工具更友好，也能准确判断出 String 或 Integer
> When try to get an error type of the value, return null instead of exception. it will be friendly to serialization tools, and can accurately determine String or Integer.

- 使用两个字段表示不同的值，在将数据推送给 Kubernetes API时，再进行判别
> Using two different fields to save two type values (one of them is null), and when push to Kubernetes API server, choose the right one.

或者，在这次修改后功能已经稳定了，不再做任何改动。
> Finally, we would do nothing after this PR if it's already stable enough

File: kubernetes/src/main/java/io/kubernetes/client/custom/IntOrStringJacksonSerializer.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (brendandburns) [inline kubernetes/src/main/java/io/kubernetes/client/custom/IntOrStringJacksonSerializer.java:2]: nit: 2024

**Comment by brendandburns** [comment]
/lgtm
/approve

Thanks for the PR, especially thanks for writing unit tests!

**Comment by brendandburns** [comment]
Sorry one last thing, can you remove the wiremock changes? I'd rather that was sent as a separate PR.

**Comment by wkclz** [comment]
The old version of wiremock dependency old Jackson and conflicts with it.
But it can update wiremock (with Jackson) and then fix IntOrString.

Do we need to split into two PR?

**Comment by yue9944882** [comment]
+1 separating into two PRs for clarity

**Comment by brendandburns** [comment]
Thanks, please send the wiremock update PR and we will merge it first, then you can rebase this PR.

**Comment by codefromthecrypt** [comment]
@wkclz thanks for sorting this out!

**Comment by wkclz** [comment]
update: update wiremock & jackson #3126

fix: Add JacksonSerializer to serialize IntOrString #3127

Then, this PR can be reject and close.

**Pull request closed without merging**